qf/validation.go

```diff
@@ -67,7 +67,7 @@ func (req *SubmissionRequest) IsValid() bool {
 
 // IsValid ensures that both submission and course IDs are set
 func (req *UpdateSubmissionRequest) IsValid() bool {
-	return req.GetCourseID() > 0 && req.GetSubmissionID() > 0
+	return req.GetCourseID() > 0 && (req.GetSubmissionID() > 0 || req.GetAssignmentID() > 0)
```

Comment:
See discussion above. Also, whatever the conclusion about the above, we should add a test for this.